### DISCUSSION TITLE
Use state.last_reported to prevent unnecessary safety mode

### DISCUSSION THREAD

**Discussion by madd0** [jmcollin78/versatile_thermostat#629] [Ideas]
It's probably because my sensors are pretty lazy, and/or because the temperature in my house is pretty stable, but my thermostats keep going into safety mode.

I went through the code rather quickly and noticed that the component uses `last_changed` and `last_updated` to determine if the sensor is still providing a temperature reading. I took a look at the [docs of the state object](https://www.home-assistant.io/docs/configuration/state_object/) and understood the following:

* `last_changed` changes only if the state changes, i.e. if the temperature changes, in this case
* `last_updated` changes if the state or any attributes change, which in the case of my temperature sensors is the same as `last_changed` as they don't really have any other attributes that change

The most interesting thing I noticed while reading the docs, though, is that there is now a `last_reported` timestamp on the state object, which changes when an integration sets the state of an entity, regardless of any change to the state or a state attribute. This was added [around March of this year](https://developers.home-assistant.io/blog/2024/03/20/state_reported_timestamp/) apparently.

I checked some of my sensors, and indeed, `last_reported` is significantly different from the other values. For example:

![image](https://github.com/user-attachments/assets/b9a9c124-91d4-490a-a16e-be50e4aa912c)

I didn't go into the code in enough detail to determine what changes are needed, but what do you think about using `last_reported` as the value used to determine temperature "freshness"? Maybe with `last_changed` as a fallback if `last_reported` is not available (old HA version)?

**Comment by jmcollin78** [comment]
Hello, yes I remember I have tried this (there should be some piece of code around this) but this was not working on my environment. Maybe it depends on the brand/model or techno (Zigbee / mqtt / wifi / ...).

I have implemented the same thing with a last_seen info if you have it. You can provide an entity that gives the last_seen datetime value. You can also map the last_seen to the reported value if it works on your home.

**Reply by madd0** [reply]
Ah, yes, that makes sense.

I've only recently become familiar with template sensors, which I've been using more and more for exactly this kind of scenario.

That's why I started a discussion rather than an outright feature request :)
